Makefile

```diff
@@ -1,38 +1,78 @@
-.PHONY: all clean testing time time2 debug debug-gcc6 data cram unit cram-local cram-external cram-integration tests tools dist
+.PHONY: all clean testing time time2 debug debug-gcc6 data cram unit cram-local cram-external cram-integration tests tools dist configure install rebuild build
 
-all: release
+all: default  # for consumers expecting to see a build/ directory, but should be build-release
 
 clean:
-	rm -rf build
+	rm -rf ${BDIR}
+	# git clean -xdf .
 
 # Custom define flags:
 #	EXPERIMENTAL_QUERY_MASK
 #	RAPTOR_TESTING_MODE
 #	RAPTOR_DEBUG_TIMINGS
 
-release:
-	@echo "[Invoking Meson]"
-	@(cd build && meson --reconfigure && ninja) || (mkdir -p build && cd build && meson --buildtype=release -DRAPTOR_TESTING_MODE=false -Dc_args=-O3 && ninja)
-
-testing:
-	@echo "[Invoking Meson]"
-	@(cd build-testing && meson --reconfigure && ninja) || (mkdir -p build-testing && cd build-testing && meson --buildtype=release -DRAPTOR_TESTING_MODE=true -Dc_args=-O3 && ninja)
-
-time:
-	@echo "[Invoking Meson]"
-	@(cd build-time && meson --reconfigure && ninja) || (mkdir -p build-time && cd build-time && meson --buildtype=release -DRAPTOR_TESTING_MODE=true -DRAPTOR_DEBUG_TIMINGS=true -Dc_args=-O3 && ninja)
-
-time2:
-	@echo "[Invoking Meson]"
-	@(cd build-time2 && meson --reconfigure && ninja) || (mkdir -p build-time2 && cd build-time2 && meson --buildtype=release -DRAPTOR_DEBUG_TIMINGS=true -Dc_args=-O3 && ninja)
-
-debug:
-	@echo "[Invoking Meson]"
-	@(cd build-debug && meson --reconfigure && ninja) || (mkdir -p build-debug && cd build-debug && (meson --buildtype=debug -Db_sanitize=address -Dc_args=-O3) && ninja)
-
-debug-gcc6:
-	@echo "[Invoking Meson]"
-	@(cd build-debug-gcc6 && meson --reconfigure && ninja) || (mkdir -p build-debug-gcc6 && cd build-debug-gcc6 && (env CC=gcc-6 CXX=g++-6 meson --buildtype=debug -Db_sanitize=address -Dc_args=-O3) && ninja)
+PREFIX?=${CURDIR}/PREFIX
+BDIR?=build
+# Most rules will create BDIR only if it does not already exist.
+# ("|" means  "order-only" rule, useful for directory creation.)
+
+rebuild: | ${BDIR}
+	ninja -C ${BDIR} reconfigure
+	ninja -C ${BDIR}
+install: | ${BDIR}
+	ninja -C ${BDIR} reconfigure
+	ninja -C ${BDIR} install
+
+# "meson --reconfigure" is not idempotent, but
+# "ninja reconfigure" works even the first time.
+
+MESON_FLAGS?=--buildtype=debug -DRAPTOR_TESTING_MODE=false -Dc_args=-O0 --prefix=${PREFIX}
+
+# This is the only rule that uses MESON_FLAGS.
+# If you want to recreate a directory, you can run "make configure", or simply "rm -rf build-dir".
+configure:
+	rm -rf ${BDIR} && mkdir -p ${BDIR} && meson ${MESON_FLAGS} ${BDIR}
```

Comment:
This doesn't look good - why would we want to delete the build directory every time a user runs `make build`? The directory is there, it can be easily reconfigured. That should be better for rebuilds, seeing how Pbbam is a massive dependency.